ml-conversational-analytic-tool/githubDataExtraction.py

```diff
@@ -161,17 +155,26 @@ def getPullFeatures(self, pull):
     parser = argparse.ArgumentParser(description='Create csv for all pulls in repo')
     parser.add_argument('organization', help='Organization the repo belongs to.')
     parser.add_argument('reponame', help='Name of repo')
-    parser.add_argument('-reactions', action='store_true', default=False, help='Flag to extract reactions')
-    parser.add_argument('--filename', help='Name of file')
-    # parser.add_argument('accesstoken', help='Github access token')
+    parser.add_argument('--reactions', action='store_true', default=False, help='Flag to extract reactions')
+    parser.add_argument('-n', "--name", help='Output file name. If not specified, the name is constructed like this: '
+            '<organization><reponame>.csv')
 
     args = parser.parse_args()
     ACCESS_TOKEN = os.environ["GITACCESS"]  # Access Github token from environment for security purposes
     extractor = GithubDataExtractor(ACCESS_TOKEN)  # Create object
-    extractor.openRepo(args.organization, args.reponame)  # Open repo
 
-    # Extract all pulls and export them to .csv
-    if args.filename:
-        extractor.getAllPulls(args.filename, args.reactions)
+    if args.reponame == 'all':
```

Comment:
What do you think about making `reponame` an optional argument and adding `--all` flag?
Instead of using reponame to pass `all`, look at the flag instead?

Reply:
Agree, `--reponame` and `--all` to be mutually exclusive  - I mean, the user should provide one of them only. 

Neet to consider that the `--name` parameter (output file name) should be applicable only in the case of the single repo.
In the case of --all repositories, the names of the output files need to be always auto-generated. (#21)

And probably it will be good to rename `reponame` to just `repo` and stick with `name` for the output file name ( just to be consistent with the other commands that use `--name` for output file name).

Reply:
So I took out the 'all' out entirely. And I made repo optional. If the user doesn't specify a repo then data will be extracted from all the repos in the organization.

Reply:
I am okay with @enyinna1234 's approach to make repo an optional flag and removing the need to explicitly state --all. WDYT @difince ?

Reply:
With my suggestion, the valid combinations are these:
```
python ./ml-conversational-analytic-tool/githubDataExtraction.py vmware-labs  --all
python ./ml-conversational-analytic-tool/githubDataExtraction.py vmware-labs  --repo ml-conversational-analytic-tool --name output_file_name
python ./ml-conversational-analytic-tool/githubDataExtraction.py vmware-labs  --repo ml-conversational-analytic-tool
```
Invalid combinations:
```
python ./ml-conversational-analytic-tool/githubDataExtraction.py vmware-labs  --all --repo <some_repo>
python ./ml-conversational-analytic-tool/githubDataExtraction.py vmware-labs  --all --name
python ./ml-conversational-analytic-tool/githubDataExtraction.py vmware-labs  --all --name --repo <some_name>
```
I see we could use:
 `ArgumentParser.add_argument_group`  or 
 `ArgumentParser.add_mutually_exclusive_group` or 
`ArgumentParser.add_subparsers` or 
just validate the inputs like:
```
args = parser.parse_args()
if ('all' in vars(args) and 'repo' in vars(args))
    raise Exception  
```
WDYT?

Reply:
I'd also encourage you to use environment variables similar to GitHub CLI as they have become somewhat of a standard. https://cli.github.com/manual/gh_help_environment.  So for repo, `GH_REPO` is used.  And as a command line argument, GitHub CLI uses
`   -R, --repo [HOST/]OWNER/REPO   Select another repository using the [HOST/]OWNER/REPO format`

And when you run the command within the directory of a git project, and no argument is passed, why not read the .gitconfig for the project to determine the repo.

Reply:
we're in an agreement of needing to restrict the use of `--name` to only when `--repo` is specified.

I think the question here is, do we need to explicitly have `--all` flag.
Can that be assumed when no flag is passed?

```
python ./ml-conversational-analytic-tool/githubDataExtraction.py vmware-labs --all
```

or

```
python ./ml-conversational-analytic-tool/githubDataExtraction.py vmware-labs
```

Reply:
The issue I see with making both --all and --repo optional is that you actually must have either or. Which means that they're really not optional. The problem is that in that case, running the command below will be valid:
`python ./ml-conversational-analytic-tool/githubDataExtraction.py vmware-labs`
And that shouldn't be a valid command because it will not extract from the organization or repo level

Reply:
In my proposal `githubDataExtraction.py vmware-labs` should not be valid. This command is short and easy, but it could be heavy and time-consuming. That's the reason I was thinking it is better to be explicit.  But still, I am OK.

Few ideas if we want to use `githubDataExtraction.py vmware-labs` to pull all repos. (could be implemented in separate PR if we agree)  - We could add a warning message in the argument help msg / or even ask for confirmation: 

> "Caution: The command will pull the data from all the repos in %s organization. It could take a long time. Are you sure you want to proceed?  Alternative: command --repo <repo_name>
Please answer with: Y (yes), No (n)
 
Erick's commend introduces a brand new idea - so better to become a proposal.

Reply:
Let's use the approach you suggested `githubDataExtraction.py vmware-labs` and keep things simple and short

Reply:
Added -R as an argument for repo. I thought about how to use an environment variable for repo per @ericwb's suggestion, and I agree with @difince, that's a new idea. Will the user change the environment variable anytime they want data from a different repo?